src/store/jobs.js

```diff
@@ -0,0 +1,135 @@
+import Vue from 'vue';
+
+function noop() {}
+
+export const JobState = {
```

Comment:
I'm not familiar with web workers but it seems like this functionality should already be available in some library. Is it worth doing that now?